virtualizarr/tests/test_writers/test_icechunk.py

```diff
@@ -0,0 +1,292 @@
+    manifest = ChunkManifest(
+        {
+            "0.0": {"path": netcdf4_file, "offset": 200, "length": 16},
+            "0.1": {"path": netcdf4_file, "offset": 216, "length": 16},
+            "1.0": {"path": netcdf4_file, "offset": 232, "length": 16},
+            "1.1": {"path": netcdf4_file, "offset": 248, "length": 16},
+        }
+    )
+    zarray = ZArray(
+        shape=(4, 4),
+        chunks=(2, 2),
+        dtype=np.dtype("<i4"),
+        compressor=None,
+        filters=None,
+        fill_value=None,
+    )
+    ma = ManifestArray(
+        chunkmanifest=manifest,
+        zarray=zarray,
+    )
+    air = Variable(data=ma, dims=["y", "x"])
+    vds = Dataset(
+        {"air": air},
+    )
+
+    dataset_to_icechunk(vds, icechunk_filestore)
+
+    root_group = group(store=icechunk_filestore)
+    air_array = root_group["air"]
+    assert isinstance(air_array, Array)
+
+    # check array metadata
+    assert air_array.shape == (4, 4)
+    assert air_array.chunks == (2, 2)
+    assert air_array.dtype == np.dtype("int32")
+
+    # check chunk references
+    assert np.allclose(
+        air_array[:2, :2], np.frombuffer(actual_data[:16], "<i4").reshape(2, 2)
+    )
+    assert np.allclose(
+        air_array[:2, 2:], np.frombuffer(actual_data[16:32], "<i4").reshape(2, 2)
+    )
+    assert np.allclose(
+        air_array[2:, :2], np.frombuffer(actual_data[32:48], "<i4").reshape(2, 2)
+    )
+    assert np.allclose(
+        air_array[2:, 2:], np.frombuffer(actual_data[48:], "<i4").reshape(2, 2)
+    )
+
+
+def test_write_loadable_variable(
+    icechunk_filestore: "IcechunkStore", simple_netcdf4: Path
+):
+    # instead for now just write out byte ranges explicitly
+    manifest = ChunkManifest(
+        {"0.0": {"path": simple_netcdf4, "offset": 6144, "length": 48}}
+    )
+    zarray = ZArray(
+        shape=(3, 4),
+        chunks=(3, 4),
+        dtype=np.dtype("int32"),
+        compressor=None,
+        filters=None,
+        fill_value=None,
+    )
+    ma = ManifestArray(
+        chunkmanifest=manifest,
+        zarray=zarray,
+    )
+
+    ma_v = Variable(data=ma, dims=["x", "y"])
+
+    la_v = Variable(
+        dims=["x", "y"],
+        data=np.random.rand(3, 4),
+        attrs={"units": "km"},
+    )
+    vds = Dataset({"air": la_v}, {"pres": ma_v})
+
+    dataset_to_icechunk(vds, icechunk_filestore)
+
+    root_group = group(store=icechunk_filestore)
+    air_array = root_group["air"]
+    assert isinstance(air_array, Array)
+    assert air_array.shape == (3, 4)
+    assert air_array.dtype == np.dtype("float64")
+    assert air_array.attrs["units"] == "km"
+    assert np.allclose(air_array[:], la_v[:])
+
+    pres_array = root_group["pres"]
+    assert isinstance(pres_array, Array)
+    assert pres_array.shape == (3, 4)
+    assert pres_array.dtype == np.dtype("int32")
+    expected_ds = open_dataset(simple_netcdf4)
+    expected_array = expected_ds["foo"].to_numpy()
+    npt.assert_equal(pres_array, expected_array)
```

Comment:
This can also all be checked by doing `open_zarr` then using a single xarray assertion. But it might be valuable to keep these lower-level assertions too.